rules-tests/TypeDeclaration/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/base_find.php.inc

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddArrayReturnDocTypeRector\Fixture;
+
+class BaseFind
+{
+    /**
+     * @return static|false
+     */
+    public function find()
+    {
+    }    
+}
+
+class Finder extends BaseFind
+{
+    public function something()
+    {
+        return $this->find();
+    }   
+}
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddArrayReturnDocTypeRector\Fixture;
+
+class BaseFind
+{
+    /**
+     * @return static|false
+     */
+    public function find()
+    {
+    }    
+}
+
+class Finder extends BaseFind
+{
+    /**
+     * @return static|false
+     */
```

Comment:
atm rector is infering a different type here and therefore looses precision

ACTUAL:
```
+    /**
+     * @return bool|$this
+     */
```

Reply:
probably related with https://github.com/rectorphp/rector-src/pull/1521